component/phlare/write/write.go

```diff
@@ -202,6 +244,30 @@ func (f *fanOutClient) Push(ctx context.Context, req *connect.Request[pushv1.Pus
 	return connect.NewResponse(&pushv1.PushResponse{}), nil
 }
 
+func shouldRetry(err error) bool {
+	if err == context.DeadlineExceeded {
+		return true
+	}
```

Comment:
Should this be `errors.Is(err, context.DeadlineExceeded)` to handle wrapped errors? If not, can you add a comment to explain why it shouldn't so we don't accidentally swap it out?